swarm/shed/field_struct.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package shed
+
+import (
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+// StructField is a helper to store complex structure by
+// encoding it in RLP format.
+type StructField struct {
+	db  *DB
+	key []byte
+}
+
+// NewStructField returns a new StructField.
+// It validates its name and type against the database schema.
+func (db *DB) NewStructField(name string) (f StructField, err error) {
+	key, err := db.schemaFieldKey(name, "struct-rlp")
+	if err != nil {
+		return f, err
+	}
+	return StructField{
+		db:  db,
+		key: key,
+	}, nil
+}
+
+// Unmarshal unmarshals data fromt he database to a provided val.
+// If the data is not found leveldb.ErrNotFound is returned.
+func (f StructField) Unmarshal(val interface{}) (err error) {
```

Comment:
are you sure we want to call this `Unmarshal`? this does a `Get` on the db so why not call it `Get`? it is also symmetric to the `Put` below

Reply:
I decided for Unmarshal as it populates the provided interface. Get is also fine and clear, maybe better.